confignode/src/main/java/org/apache/iotdb/confignode/procedure/env/DataNodeRemoveHandler.java

```diff
@@ -135,29 +136,99 @@ public TDataNodeLocation findDestDataNode(TConsensusGroupId regionId) {
   }
 
   /**
-   * Send to DataNode, migrate region from originalDataNode to destDataNode
+   * Send to DataNode, add region peer
    *
    * @param originalDataNode old location data node
    * @param destDataNode dest data node
    * @param regionId region id
    * @return migrate status
    */
-  public TSStatus migrateRegion(
+  public TSStatus addRegionPeer(
       TDataNodeLocation originalDataNode,
       TDataNodeLocation destDataNode,
       TConsensusGroupId regionId) {
     TSStatus status;
-    List<TDataNodeLocation> regionReplicaNodes = findRegionReplicaNodes(regionId);
-    if (regionReplicaNodes.isEmpty()) {
-      LOGGER.warn("Not find region replica nodes, region: {}", regionId);
+    Optional<TDataNodeLocation> newLeaderNode = findNewLeader(regionId, originalDataNode);
+    if (!newLeaderNode.isPresent()) {
+      LOGGER.warn(
+          "No other Node to change region leader, check by show regions, region: {}", regionId);
       status = new TSStatus(TSStatusCode.MIGRATE_REGION_ERROR.getStatusCode());
-      status.setMessage("not find region replica nodes, region: " + regionId);
+      status.setMessage("No other Node to change region leader, check by show regions");
       return status;
     }
 
-    // TODO if region replica is 1, the new leader is null, it also need to migrate
-    Optional<TDataNodeLocation> newLeaderNode =
-        regionReplicaNodes.stream().filter(e -> !e.equals(originalDataNode)).findAny();
+    TMigrateRegionReq migrateRegionReq =
+        new TMigrateRegionReq(regionId, destDataNode, destDataNode);
```

Comment:
Why the parameter _destDataNode_ occurs twice here?

Reply:
this does not need originalDataNode, and can not be null. so use destDataNode fills.